internal/factsengine/gatherers/hostsfile.go

```diff
@@ -0,0 +1,74 @@
+package gatherers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/trento-project/agent/internal/utils"
+)
+
+const (
+	HostsFactKey  = "hosts"
+	HostsFilePath = "/etc/hosts"
+)
+
+type HostsFileGatherer struct {
+	FileSystem afero.Fs
+}
+
+func NewHostsFileGatherer(fileSystem afero.Fs) *HostsFileGatherer {
+	return &HostsFileGatherer{FileSystem: fileSystem}
+}
+
+func (s *HostsFileGatherer) Gather(factsRequests []FactRequest) ([]Fact, error) {
+	facts := []Fact{}
+	log.Infof("Starting /etc/hosts file facts gathering process")
+
+	hostsFile, err := s.FileSystem.Open(HostsFilePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not open /etc/hosts file")
+	}
+
+	defer hostsFile.Close()
+
+	hostsFileContents, err := ioutil.ReadAll(hostsFile)
+	if err != nil {
+		return facts, err
+	}
+
+	hostsMap := utils.FindMatches(`(?m)^([^#|\s]+)\s([^#\n]+)#?.*$`, hostsFileContents)
+
+	for _, factReq := range factsRequests {
+		factValue := searchMapByValue(hostsMap, factReq.Argument)
+		if len(factValue) < 1 {
+			continue
```

Comment:
Is this the piece of code where we should return an failed fact? Could we put a `TODO` comment?

Reply:
I wouldn't add a TODO there as I think it could be an option to just return no results for a fact request. But we need to discuss that and come back to this.